src/library/scala/Array.scala

```diff
@@ -513,7 +513,7 @@ object Array {
   def equals(xs: Array[AnyRef], ys: Array[AnyRef]): Boolean = {
     if (xs eq ys)
       return true
-    if (xs.length != ys.length)
+    if (xs.lengthCompare(ys.length) != 0)
```

Comment:
I don’t think you save anything with this change. `lengthCompare` is implemented in terms of comparing `.length`.

Reply:
@julienrf I agree. I'll clean it up. it has already been concluded that these changes will probably improve the work only of the heirs to LinearSeq